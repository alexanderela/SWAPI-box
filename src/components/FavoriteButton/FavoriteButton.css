.FavoriteButton {
  margin: 0;
}

.favorites-button {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.favorites-button:hover {
  background: #f7d9f7;
  box-shadow: #f7d9f7 0px 0px 20px 2px;
  text-shadow: 
    #c0c 0px 0px 6px, 
    #c0c 0px 0px 10px;
  width:200%;
}


