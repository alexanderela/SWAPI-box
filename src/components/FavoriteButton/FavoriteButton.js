import React from 'react';
import './FavoriteButton.css';
import PropTypes from 'prop-types';

const FavoriteButton = ({favorites, showFavorites, checkIfAnyFavorites}) => {
	return(
		<div className="FavoriteButton">
	    <button 
	    	className="cat-button favorites-button"
	    	// onClick={() => showFavorites(favorites)}
	    	onClick={() => checkIfAnyFavorites(favorites)}
	    	>
	      Favorites
	      	<span 
	      		className="favorites-count">
	      		{favorites ? favorites.length : 0}
	      	</span>
	    </button>
		</div>
	)
}

FavoriteButton.propTypes = {
	favoritesCount: PropTypes.oneOfType([
  PropTypes.array,
  PropTypes.number
  ]),
	checkIfAnyFavorites: PropTypes.func
}

export default FavoriteButton;
