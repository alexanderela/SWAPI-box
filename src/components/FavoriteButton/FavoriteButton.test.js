import React from 'react';
import ReactDOM from 'react-dom';
import FavoriteButton from './FavoriteButton';
import { shallow, mount } from 'enzyme';

describe('FavoriteButton', () => {
	
	it('matches the snapshot', () => {
		let mockFavoriteCount = []
		let wrapper = shallow(<FavoriteButton favoritesCount={[mockFavoriteCount]}/>);
		expect(wrapper).toMatchSnapshot();
	})
})
