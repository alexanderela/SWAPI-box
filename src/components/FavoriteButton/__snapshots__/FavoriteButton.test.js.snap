// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteButton
    favoritesCount={
      Array [
        Array [],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="cat-button favorites-button"
      >
        Favorites
        <span
          className="favorites-count"
        />
      </button>,
      "className": "FavoriteButton",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Favorites",
          <span
            className="favorites-count"
          />,
        ],
        "className": "cat-button favorites-button",
      },
      "ref": null,
      "rendered": Array [
        "Favorites",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
            ],
            "className": "favorites-count",
          },
          "ref": null,
          "rendered": Array [],
          "type": "span",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="cat-button favorites-button"
        >
          Favorites
          <span
            className="favorites-count"
          />
        </button>,
        "className": "FavoriteButton",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites",
            <span
              className="favorites-count"
            />,
          ],
          "className": "cat-button favorites-button",
        },
        "ref": null,
        "rendered": Array [
          "Favorites",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [],
              ],
              "className": "favorites-count",
            },
            "ref": null,
            "rendered": Array [],
            "type": "span",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
